### DISCUSSION TITLE
Taylor Factor from Finite Element Strain Tensor Data

### DISCUSSION THREAD

**Discussion by Mustafa-Rifat** [mtex-toolbox/mtex#777] [Ask Anything] [answered]
Hello Mtex community,

I want to find the taylor factor plot from my ebsd plot. I have performed nanoscratch tests, and the scratches pass through a number of grains. I have modeled the scratch test using Finite Element (FE) method. I have taken an element of interest from the FE model and extracted the strain tensor from it. Now I am using that to figure out the Taylor factor plot.

I just want to be sure that I am doing it right.  At first I import and plot the ebsd data. I am ignoring that part here. After that the code is:
```
grains = calcGrains(ebsd('Ni-superalloy'));
grains = smooth(grains,5);
%strain_FEA is the strain tensor I get from FE model
strain_FEA=[0.00958486 -0.00971106 0.043623; -0.00971106 0.0278111 0.0199594; 0.043623 0.0199594 -0.037396];
%normalizing the strain tensor
normalized_strain_FEA=normalize(strain_FEA);
epsilon = strainTensor(normalized_strain_FEA);
sS = symmetrise(slipSystem.fcc(grains.CS));
[M,b,W] = calcTaylor(inv(grains.meanOrientation)*epsilon,sS);
figure
plot(grains,M,'micronbar','off')
mtexColorMap white2black
mtexColorbar
```
The plot and values of taylor factor I get from this are reasonable (>3 which is true for fcc materials). 
![image](https://user-images.githubusercontent.com/60717121/107991377-7a2cad00-6fa4-11eb-8d99-b2355e3d9d34.png)
What I understand from the code is to apply calcTaylor(eps,sS), the eps and sS both should be in crystal reference frame (Link: https://mtex-toolbox.github.io/strainTensor.calcTaylor.html). Now the strain tensor I get from the FE model is for specimen reference frame. The inv(grains.meanOrientation)*epsilon inside calcTaylor(inv(grains.meanOrientation)*epsilon,sS) expression takes the epsilon from specimen reference frame to crystal reference frame and that is what we need. Please correct me if I am wrong. May be I am thinking it too simply.

**Comment by kilir** [comment, marked as answer]
Hi,
I think for the normalization, you should divide the strain tensor by its Frobenius norm:
```
strain_FEA=[0.00958486 -0.00971106 0.043623; -0.00971106 0.0278111 0.0199594; 0.043623 0.0199594 -0.037396];
epsilon = strainTensor(normalized_strain_FEA);
%normalizing the strain tensor
epsilon = epsilon/epsilon.norm
```
One thing to take care of, is, that the x,y,z coordinate systems (from your FE model and in mtex) do properly coincide. I guess you have checked that.

Cheers,
Rüdiger

**Reply by Mustafa-Rifat** [reply]
Hi Rüdiger,

When I try to do epsilon.norm I get the following error:
No appropriate method, property, or field 'norm' for class 'strainTensor'.

```
Error in tensor/subsref (line 58)
  [varargout{1:nargout}] = builtin('subsref',T,s);
```
I use mtex 5.1.1.

**Reply by kilir** [reply]
Hi,

tensor/norm() was introduced about 29 month ago, using a more recent version may be a solution - and will potentially have many known bugs fixed. Alternatively just normalize the matrix properly `Ea/norm(Ea.matrix,'fro')`.

Cheers,
Rüdiger

**Reply by Mustafa-Rifat** [reply]
Hello,

I followed what you suggested. 
```
strain_FEA=[0.00958486 -0.00971106 0.043623; -0.00971106 0.0278111 0.0199594; 0.043623 0.0199594 -0.037396];
normalized_strain_FEA=strain_FEA./norm(strain_FEA,'fro');
epsilon = strainTensor(normalized_strain_FEA);
```
Rest are the same. Now what I get is like this:
![image](https://user-images.githubusercontent.com/60717121/108112525-36997800-7064-11eb-8247-fd9b3a10ffc5.png)

Only thing that is bothering me is for FCC materials, Taylor Factor value is ~3. Here my average is about 2.5. Is it wrong?

**Reply by kilir** [reply]
The numeric value of M depends on the normalization, if you divide your strain tensor by 10 it will be 10 times smaller.
```
cs = crystalSymmetry('m-3m');
sS = symmetrise(slipSystem.fcc(cs));

o = regularSO3Grid(cs);
o = o(:);

eps = strainTensor([1 0 0; 0 -1 0; 0 0 0])
[M,~,W] = calcTaylor(inv(o)*eps,sS);

mean(M)/norm(eps.matrix,'fro')
mean(M)/norm(eps.matrix)
```

**Reply by Mustafa-Rifat** [reply]
Thanks a lot!
Just one last query, after I divided the strain tensor from FE model by its frobenius norm and recalculated the strain tensor, this is how it looks:
```
epsilon
 
epsilon = strainTensor (show methods, plot)
  rank: 2 (3 x 3)
 
 *10^-2
  11.41 -11.56  51.94
 -11.56  33.11  23.76
  51.94  23.76 -44.52
```
i.e. this is not something simple as uni/biaxial (like [1 0 0;0 -1 0;0 0 0]), still the work that I am doing is valid, right?

My model is for a scratch test, which is definitely not uniaxial so I am not expecting to see anything like that too. However, in another thread there was a conversation which made me a little bit confused.
Thank you!
